src/main/java/com/softserve/service/impl/TeacherServiceImpl.java

```diff
@@ -220,4 +234,160 @@ private void updateEmailInUserForTeacher(String email, long userId) {
     private boolean isEmailNullOrEmpty(String email) {
         return email == null || email.isEmpty();
     }
+
+    /**
+     * Each line of the file should consist of five fields, separated by commas without spaceBar.
+     * First line of the file is a header.
+     * All subsequent lines contain data about teachers.
+     * <p>
+     * name,surname,patronymic,position,email
+     * Test1,Test1,Test1,test1,test1@gmail.com
+     * Test2,Test2,Test2,test2,test2@gmail.com
+     * etc.
+     * <p>
+     *
+     * @param file file with teachers data
+     * @return list of created teachers.
+     */
+    @Override
+    public List<TeacherImportDTO> saveFromFile(MultipartFile file, Long departmentId) {
+        log.info("Enter into saveFromFile of TeacherServiceImpl");
+
+        List<TeacherImportDTO> teachers = CsvFileParser.getTeachersFromFile(file);
+
+        List<TeacherImportDTO> savedTeachers = new ArrayList<>();
```

Comment:
Lines 258 - 265 can be replaced by:

return teachers.stream().map(teacher -> saveTeacher(departmentId, teacher)).collect(Collectors.toList());

better? :)

Reply:
Much :D